### PR TITLE
Hooks

### PR DESCRIPTION
**Pull request by MarinaStepanchuk**
Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module03
Screenshot:
Screenshot_31

Screenshot_32

Screenshot_33

Screenshot_34

Deploy: [movie-search-rs.netlify.app](https://movie-search-rs.netlify.app/)
Done 01.04.2023 / deadline 03.04.2023
Score: 15 / 15

### CODE REVIEW

**Review by Ksarelto — COMMENTED** [review]
please, take a look on props destructure, return types in all code

File: src/components/ItemMovie/ItemMovie.tsx [hunk 1]
```diff
@@ -9,48 +8,46 @@ interface IItemMoviesProps {
...
 
-  public render(): JSX.Element | undefined {
-    const { name, poster, year, countries, rating } = this.props.movie;
+  const { name, poster, year, countries, rating } = props.movie;
 
-    if (!this.foundMovie(this.props.movie)) {
-      return;
-    }
+  if (!foundMovie(props.movie)) {
+    return <></>;
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/ItemMovie/ItemMovie.tsx:27]: corrected

File: src/constants/common.constants.tsx [hunk 2]
```diff
@@ -317,9 +317,10 @@ const ErrorMessages = {
   missingPhoto: 'Missing photo',
   wrongDate: 'It is not possible to choose a date longer than today',
   recommended: 'It is necessary to leave recommendations',
+  personalData: 'You must consent to the processing of personal data',
 };
 
-const regName = /^([a-zA-Z]{3,})[ ]([a-zA-Z]{3,})(([a-zA-Z]{0,}){0,})/gm;
+const regName = /^([A-Z А-Я][a-zа-я]{2,})[ ]([A-Z А-Я][a-zа-я]{2,})(([A-Z А-Я][a-zа-я]{0,}){0,})/gm;
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/constants/common.constants.tsx:323]: renamed

File: src/containers/ListCritiques/ListCritiques.tsx [hunk 3]
```diff
@@ -3,16 +3,14 @@ import { ICritique } from '../../types/interfaces';
...
-      <div className={classes.list}>
-        {this.props.critiques.map((critique, index) => (
-          <ItemCritique key={index} critique={critique} />
-        ))}
-      </div>
-    );
-  }
-}
+const ListCritiques = (props: { critiques: Array<ICritique> }) => {
+  return (
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/containers/ListCritiques/ListCritiques.tsx:7]: corrected

File: src/containers/Movies/Movies.tsx [hunk 4]
```diff
@@ -1,53 +1,30 @@
...
 import ListMovies from '../ListMovies/ListMovies';
-import { IMovie } from '../../types/interfaces';
 import { LocalStorageKeys } from '../../constants/common.constants';
 
-interface IListMoviesProps {
-  movies: IMovie[];
-  text: string;
-}
+const Movies = (): JSX.Element => {
+  const [movies] = useState(defaultMoviesState);
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/containers/Movies/Movies.tsx:8]: changed to constant

File: src/containers/Movies/Movies.tsx [hunk 5]
```diff
@@ -1,53 +1,30 @@
...
-    return (
-      <section>
-        <SearchBar text={this.state.text} onTextChange={this.handleTextChange} />
-        <ListMovies text={this.state.text} movies={this.state.movies} />
-      </section>
-    );
-  }
-}
+  useEffect(() => {
+    localStorage.setItem(LocalStorageKeys.search, `${text}`);
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/containers/Movies/Movies.tsx:16]: great idea)

File: src/containers/Movies/Movies.tsx [hunk 6]
```diff
@@ -1,53 +1,30 @@
...
-  public componentWillUnmount(): void {
-    window.removeEventListener('beforeunload', this.saveText);
-    this.saveText();
-  }
-
-  private handleTextChange: (text: string) => void = (text: string) => {
-    this.setState({
-      text: text,
-    });
+  const handleTextChange: (text: string) => void = (text: string) => {
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/containers/Movies/Movies.tsx:11]: this looks much better, thanks)

File: src/components/ItemMovie/ItemMovie.tsx [hunk 7]
```diff
@@ -9,48 +8,46 @@ interface IItemMoviesProps {
...
       !year.toString().includes(text) &&
       !rating.toString().includes(text) &&
       countries.filter((country) => country.name.includes(text)).length === 0;
     return !movieFound;
-  }
+  };
 
-  public render(): JSX.Element | undefined {
-    const { name, poster, year, countries, rating } = this.props.movie;
+  const { name, poster, year, countries, rating } = props.movie;
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/ItemMovie/ItemMovie.tsx:24]: endured.

File: src/components/ItemMovie/ItemMovie.tsx [hunk 8]
```diff
@@ -9,48 +8,46 @@ interface IItemMoviesProps {
   text: string;
 }
 
-class ItemMovie extends React.Component<IItemMoviesProps> {
-  private foundMovie(movie: IMovie): boolean {
+const ItemMovie = (props: IItemMoviesProps) => {
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/ItemMovie/ItemMovie.tsx:11]: corrected

File: src/components/Header/Header.tsx [hunk 9]
```diff
@@ -1,69 +1,52 @@
-import React from 'react';
 import { Pages, Paths } from '../../constants/common.constants';
 import { NavLink } from 'react-router-dom';
 import getRouteTitle from '../../utils/getRouteTitle';
 import classes from './Header.module.scss';
+import { useState } from 'react';
 
-class Header extends React.Component<object, { title?: string }> {
-  public state;
+const Header = () => {
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/Header/Header.tsx:7]: added

File: src/components/Header/Header.tsx [hunk 10]
```diff
@@ -1,69 +1,52 @@
...
+  const [title, setTitle] = useState(getRouteTitle());
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/Header/Header.tsx:8]: ok. added

File: src/components/ItemCritique/ItemCritique.tsx [hunk 11]
```diff
@@ -1,46 +1,41 @@
-import React from 'react';
 import { Content } from '../../constants/common.constants';
 import { ICritique } from '../../types/interfaces';
 import classes from './ItemCritique.module.scss';
 
-class ItemCritique extends React.Component<{ critique: ICritique }> {
-  public render(): JSX.Element | undefined {
-    const { name, country, photo, date, movie, review, recommended } = this.props.critique;
+const ItemCritique = (props: { critique: ICritique }) => {
+  const { name, country, photo, date, movie, review, recommended } = props.critique;
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/ItemCritique/ItemCritique.tsx:6]: corrected. thanks

File: src/containers/ListCritiques/ListCritiques.tsx [hunk 12]
```diff
@@ -3,16 +3,14 @@ import { ICritique } from '../../types/interfaces';
...
+    <div className={classes.list}>
+      {props.critiques.map((critique, index) => (
+        <ItemCritique key={index} critique={critique} />
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/containers/ListCritiques/ListCritiques.tsx:10]: changed to name.

File: src/App/App.tsx [hunk 13]
```diff
@@ -8,20 +7,18 @@ import Header from '../components/Header/Header';
...
-          <Route path={Paths.main.path} element={<Main />} />
-          <Route path={Paths.about.path} element={<AboutUs />} />
-          <Route path={Paths.critiques.path} element={<Critiques />} />
-          <Route path={Paths.error.path} element={<Error />} />
-        </Routes>
-      </>
-    );
-  }
-}
+const App = () => {
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/App/App.tsx:10]: added

File: src/App/App.tsx [hunk 14]
```diff
@@ -8,20 +7,18 @@ import Header from '../components/Header/Header';
...
+  return (
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/App/App.tsx:11]: corrected

File: src/components/Form/Form.test.tsx [hunk 15]
```diff
@@ -1,62 +1,18 @@
...
+    render(<Form onSubmit={onSubmit} />);
   });
 
   it('should render correct', () => {
     const textBoxes = screen.getAllByRole('textbox');
     expect(textBoxes[0].getAttribute('id')).toBe(InputName.name);
     expect(screen.getByRole('checkbox').getAttribute('checked')).toBeFalsy();
   });
-
-  it('should show errors', () => {
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/Form/Form.test.tsx:24]: current conditions do not match the tests

File: src/components/FormItemWrapper/FormItemWrapper.tsx [hunk 16]
```diff
@@ -9,24 +9,22 @@ interface IFormItemProps {
   class?: string;
 }
 
-class FormItemWrapper extends React.Component<IFormItemProps> {
-  render(): JSX.Element | undefined {
-    const { label, name, children, errorMessage } = this.props;
+const FormItemWrapper = (props: IFormItemProps) => {
+  const { label, name, children, errorMessage } = props;
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/FormItemWrapper/FormItemWrapper.tsx:13]: changed

File: src/components/FormItemWrapper/FormItemWrapper.tsx [hunk 17]
```diff
@@ -9,24 +9,22 @@ interface IFormItemProps {
...
-        <div className={classes.inputContainer}>
-          <label htmlFor={name}>{label}</label>
-          {children}
-        </div>
-        {this.props.errorMessage && (
-          <p data-testid="errorMessage" className={classes.errorMessage}>
-            {errorMessage}
-          </p>
-        )}
+  return (
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/FormItemWrapper/FormItemWrapper.tsx:15]: corrected

File: src/pages/Critiques/Critiques.tsx [hunk 18]
```diff
@@ -1,131 +1,33 @@
...
-  private sendReview: (card: ICritique) => void = (card: ICritique) => {
-    this.changeReviews(card);
-  };
+const Critiques = () => {
+  const [showMessage, setShowMessage] = useState(false);
+  const [reviews, setReviews] = useState<ICritique[]>([]);
 
-  private changeReviews = (card: ICritique) => {
-    const { isValid, errorsObject } = this.isValid(card);
+  const sendReview: (card: ICritique) => void = (card) => {
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/pages/Critiques/Critiques.tsx:14]: changed

File: src/components/Form/Form.tsx [hunk 19]
```diff
@@ -1,225 +1,220 @@
...
-        : '',
-      date: this.extractStringValue(this.dateRef),
-      movie: this.extractStringValue(this.movieRef),
-      review: this.extractStringValue(this.reviewRef),
-      recommended: this.extractCheckValue(this.recommendedRef),
-      unrecommended: this.extractCheckValue(this.unrecommendedRef),
-      personal: this.extractCheckValue(this.personalRef),
+      name: name,
+      country: country,
+      photo: URL.createObjectURL(photo[0] as Blob | MediaSource),
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/Form/Form.tsx:56]: Thanks for the advice, will keep it in mind for the future

File: src/components/Header/Header.tsx [hunk 20]
```diff
@@ -1,69 +1,52 @@
...
+              onClick={() => changeTitle(Paths.main.title)}
+            >
+              {Pages.main}
+            </NavLink>
+          </li>
+          <li>
+            <NavLink
+              className={({ isActive }) => (isActive ? `${classes.activeLink}` : `${classes.link}`)}
+              to={Paths.about.path}
+              onClick={() => changeTitle(Paths.about.title)}
```
[the remark this answers is not in the document]
Comment (MarinaStepanchuk) [reply src/components/Header/Header.tsx]: corrected

**Pull request merged**